crates/chia-puzzles/src/puzzles/offer.rs

```diff
@@ -45,11 +46,11 @@ pub const SETTLEMENT_PAYMENTS_PUZZLE_V1: [u8; 267] = hex!(
 ///
 /// **Warning:**
 /// It is recommended not to use settlement payments v1 for anything other than backwards compatibility (e.g. offer compression).
-pub const SETTLEMENT_PAYMENTS_PUZZLE_HASH_V1: [u8; 32] = hex!(
+pub const SETTLEMENT_PAYMENTS_PUZZLE_HASH_V1: TreeHash = TreeHash::new(hex!(
```

Comment:
what's the motivation behind changing the type to `TreeHash`?
Is it still possible to call `.to_bytes()` on a constant, given that `to_bytes()` consumes `self`?

Reply:
Preventing mistakes with passing `[u8; 32]` instead of `TreeHash` in a generic inner puzzle field

It seems like taking self as a reference works at compile time too, for what it's worth